src/en/mangamo/src/eu/kanade/tachiyomi/extension/en/mangamo/MangamoHelper.kt

```diff
@@ -0,0 +1,80 @@
+package eu.kanade.tachiyomi.extension.en.mangamo
+
+import eu.kanade.tachiyomi.extension.en.mangamo.dto.ChapterDto
+import eu.kanade.tachiyomi.extension.en.mangamo.dto.DocumentDto
+import eu.kanade.tachiyomi.extension.en.mangamo.dto.DocumentDtoInternal
+import eu.kanade.tachiyomi.extension.en.mangamo.dto.DocumentSerializer
+import eu.kanade.tachiyomi.extension.en.mangamo.dto.SeriesDto
+import eu.kanade.tachiyomi.source.model.SManga
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.modules.SerializersModule
+import kotlinx.serialization.serializer
+import okhttp3.Headers
+import okhttp3.HttpUrl.Companion.toHttpUrl
+
+class MangamoHelper(headers: Headers) {
+
+    companion object {
+
+        val json = Json {
+            isLenient = true
+            ignoreUnknownKeys = true
+            explicitNulls = false
+
+            serializersModule = SerializersModule {
+                contextual(DocumentDto::class) { DocumentSerializer(DocumentDtoInternal.serializer(it[0])) }
+            }
+        }
+
+        @Suppress("UNCHECKED_CAST")
+        inline fun <reified T> String.parseJson(): T {
+            return when (T::class) {
+                DocumentDto::class -> json.decodeFromString<T>(
+                    DocumentSerializer(serializer<T>() as KSerializer<out DocumentDto<out Any?>>) as KSerializer<T>,
+                    this,
+                )
```

Comment:
I belive you can add ` @Serializable(with = DocumentSerializer::class)` to `DocumentDto` override the default serializer

Reply:
That was what I had initially tried, but unfortunately it didn't work because `DocumentDto` is generic and `JsonTransformingSerializer` doesn't play very nicely with that. I spent quite a while trying to get serialization to work properly before I ended up with this solution.

Reply:
looks like you are only using `SeriesDto` as type argument for `DocumentDto`, so should be fine to simplify it

Reply:
It's also used by `QueryResultDto<T>`, which is used for both chapters and series. As of the most recent change, `DocumentDto` is also being directly used on `UserDto`.